### ISSUE TITLE
thoughts

### ISSUE THREAD

**Issue by mustafa0x** [abdel-17/svelte-file-tree#1]
# p0
- [ ] actions (rename/move/copy/delete) (requires handling update events, sending them to a server, and verifying that they succeeded)
  - context menu (related: actions modal, opened with kbd)

# p1
- [ ] lazy loading
- [ ] columns (ui-only feature) (unless we add sort by columns)

# p2
- [ ] mouse interaction
  - mouse shift click (or meta+click) to select multiple
  - dnd
- [ ] jump to (separate; in modal window)
- [ ] searching/filtering (not planned)

# references
- https://github.com/lukasbach/react-complex-tree, https://rct.lukasbach.com/storybook
- https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
- https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/
- https://react-spectrum.adobe.com/react-aria/Tree.html
- https://react-arborist.netlify.app/gmail
- https://www.npmjs.com/package/react-folder-tree

**Comment by abdel-17** [comment]
Yeah shift click with the mouse is a bit more complicated because it selects everything up to the clicked item. I'll have to think about it more.

What's a "page of items" in this context?

**Comment by mustafa0x** [comment]
> What's a "page of items" in this context?

Typically it's the number of items that fit in the viewport.

**Comment by abdel-17** [comment]
That's a bit of a strange interaction to me. Do you have any examples of libraries that do this?

**Comment by mustafa0x** [comment]
https://react-spectrum.adobe.com/react-aria/ListBox.html#asynchronous-loading

![image](https://github.com/abdel-17/svelte-virtual-tree/assets/1473102/424be9fa-3be7-451d-a3e8-6d0fa7252e93)

**Comment by abdel-17** [comment]
p0 is done

**Comment by abdel-17** [comment]
> select multiple out-of-sequence. finder doesn't support this. explorer does. commander one (a util for mac) also does.

done

**Comment by abdel-17** [comment]
Implemented renaming with Enter. I noticed we discussed earlier which key should trigger "open file" functionality. The space key can be used for that. Ctrl/Option + Space toggles selection, but regular Space doesn't do anything.

**Comment by abdel-17** [comment]
I'll be closing this issue in favor of multiple issues for each feature
